File: Level1/완주하지 못한 선수.py

participant = ["mislav", "stanko", "mislav", "ana"]
completion = ["stanko", "ana", "mislav"]


def solution(participant, completion):

    participant.sort()
    completion.sort()

    for p, c in zip(participant, completion):
        if p != c:
            return p

    return p[-1]



did = solution(participant, completion)
print(did)

File: Level1/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> marathon = new HashMap<>();
        for(String par : participant) 
            marathon.put(par, 0);
        for(String fin : participant) {
            int tmp = marathon.get(fin);
            marathon.put(fin, tmp+1);
        }
        
        String answer = "";
        for(String fin : completion) {
            int tmp = marathon.get(fin);
            marathon.put(fin, tmp-1);
        }
        
        ArrayList<String> keys = new ArrayList<>(marathon.keySet());
        for(String key : keys) {
            int tmp = marathon.get(key);
            if(tmp > 0) {
                answer = key;
                break;
            }   
        }
        
        return answer;
    }
}
